src/kaitai_spec/attribute/contents.rs

```diff
@@ -58,5 +58,27 @@ impl<'de> Deserialize<'de> for Bytes {
 }
 
 pub fn try_build<'de, A: MapAccess<'de>>(context: Context) -> Either<Result<Attribute, A::Error>, ContextNoContents> {
-    todo!()
+    if let Some(Either::Left(Bytes(contents))) = &context.type_attributes {
+        if let Some(id) = context.string_keys.get("id") {
+            let attribute = Attribute {
+                id: id.clone(),
+                doc: context.string_keys.get("doc").cloned(),
+                doc_ref: context.string_keys.get("doc-ref").cloned(),
+                type_: AttributeType::Contents(contents.clone()),
+            };
+
+            return Either::Left(Ok(attribute));
+        }
+    }
+
+    Either::Right(ContextNoContents {
+        string_keys: context.string_keys,
+        size: context.size,
+        type_: context.type_,
+        type_attributes: context.type_attributes.map(|attr| match attr {
+            Either::Left(_) => Either::Right(TypeAttributes { terminator: None }),
+            Either::Right(Either::Left(enum_attr)) => Either::Left(enum_attr),
+            Either::Right(Either::Right(type_attrs)) => Either::Right(type_attrs),
+        }),
+    })
```

Comment:
here you have to check that `type_` is `None`, `size` is `None` too and process `string_keys` before constructing the result. Use constants for keys from the parent module.

Also `remove` method of `HashMap` is useful in the case: `doc: context.string_keys.remove(KEY_DOC),`